osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -729,51 +734,51 @@ private void updateYPositions()
         {
             visibleItems.Clear();
 
-            float currentY = visibleHalfHeight;
+            // visibleHalfHeight includes BleedTop, but Scroll screenspace doesn't, so we remove it here
+            float currentY = visibleHalfHeight - BleedTop;
 
             scrollTarget = null;
 
-            foreach (CarouselItem item in root.Children)
+            double elapsed = Clock.CurrentTime - timeSinceLastYUpdate;
+
+            foreach (CarouselBeatmapSet set in beatmapSets)
             {
-                if (item.Filtered.Value)
+                // find the current Y position even for filtered sets. This looks better for when the set becomes visible again as it then needs a good direction to travel from
+                set.ExpectedYPosition = timeSinceLastYUpdate == 0 ? currentY : DrawableCarouselBeatmapSet.YLerp(set.CarouselYPosition, set.ExpectedYPosition, elapsed);
```

Comment:
what is this 0 check for? is it for checking if `timeSinceLastUpdate` has been set before? if so, then use a `double?` instead of representing that information in-band like this.